finetrainers/ltx_video/ltx_video_lora.py

```diff
@@ -165,7 +221,90 @@ def prepare_latents(
         }
 
 
-def post_latent_preparation(
+def prepare_latents_i2v(
+    vae: AutoencoderKLLTXVideo,
+    image_or_video: torch.Tensor,
+    patch_size: int = 1,
+    patch_size_t: int = 1,
+    device: Optional[torch.device] = None,
+    dtype: Optional[torch.dtype] = None,
+    generator: Optional[torch.Generator] = None,
+    precompute: bool = False,
+) -> torch.Tensor:
+    device = device or vae.device
+
+    if image_or_video.ndim == 4:
+        image_or_video = image_or_video.unsqueeze(2)
+    assert image_or_video.ndim == 5, f"Expected 5D tensor, got {image_or_video.ndim}D tensor"
+
+    if image_or_video.size(2) == 1:
+        logger.warning("Received a single frame for LTX Video I2V training. Duplicating the frame to create a video.")
+        # Duplicate the image to create atleast one frame to run prediction on
+        image_or_video = image_or_video.repeat(1, 1, 2, 1, 1)
+
+    image_or_video = image_or_video.to(device=device, dtype=vae.dtype)
+    image_or_video = image_or_video.permute(0, 2, 1, 3, 4).contiguous()  # [B, C, F, H, W] -> [B, F, C, H, W]
+
+    # Note: we separately encode the image and video because there is a 4x compression applied. We only want to condition
+    # on the first frame of the video, and not the first 4 frames.
+    image, video = image_or_video[:, :, :1], image_or_video[:, :, 1:]
+
+    if not precompute:
+        image_latents = vae.encode(image).latent_dist.sample(generator=generator)
```

Comment:
Not important for this PR but just sharing info. 

Some works like InstructPix2Pix compute a mode, instead on the condition.